### PR TITLE
samba 3.6.25

### PR DESCRIPTION
**Pull request by DomT4**
Security release to fix CVE-2015-0240.

### CODE REVIEW

**Comment by DomT4** [comment]
It dumps stuff in ` lib `:

```
Installing non-libraries to "lib" is discouraged.
The offending files are:
  /usr/local/Cellar/samba/3.6.25/lib/de.msg
  /usr/local/Cellar/samba/3.6.25/lib/en.msg
  /usr/local/Cellar/samba/3.6.25/lib/fi.msg
  /usr/local/Cellar/samba/3.6.25/lib/fr.msg
  /usr/local/Cellar/samba/3.6.25/lib/it.msg
  /usr/local/Cellar/samba/3.6.25/lib/ja.msg
  /usr/local/Cellar/samba/3.6.25/lib/libnetapi.dylib.0
  /usr/local/Cellar/samba/3.6.25/lib/libsmbclient.dylib.0
  /usr/local/Cellar/samba/3.6.25/lib/libsmbsharemodes.dylib.0
  /usr/local/Cellar/samba/3.6.25/lib/libtalloc.dylib.2
  /usr/local/Cellar/samba/3.6.25/lib/libtalloc.dylib.2.0.5
  /usr/local/Cellar/samba/3.6.25/lib/libtdb.dylib.1
  /usr/local/Cellar/samba/3.6.25/lib/libtdb.dylib.1.2.9
  /usr/local/Cellar/samba/3.6.25/lib/libtevent.dylib.0
  /usr/local/Cellar/samba/3.6.25/lib/libtevent.dylib.0.9.11
  /usr/local/Cellar/samba/3.6.25/lib/lowcase.dat
  /usr/local/Cellar/samba/3.6.25/lib/nl.msg
  /usr/local/Cellar/samba/3.6.25/lib/pl.msg
  /usr/local/Cellar/samba/3.6.25/lib/ru.msg
  /usr/local/Cellar/samba/3.6.25/lib/tr.msg
  /usr/local/Cellar/samba/3.6.25/lib/upcase.dat
  /usr/local/Cellar/samba/3.6.25/lib/valid.dat
```

Are we cool with that or do we want to ` libexec ` it?

**Comment by mikemcquaid** [comment]
Did the previous release do that? Would be good to file an issue at least for the dylibs being named wrongly.

**Comment by vszakats** [comment]
Yes, previous release did this, too.

**Comment by DomT4** [comment]
I've [filed an issue upstream](https://bugzilla.samba.org/show_bug.cgi?id=11113) with them about the broken OS X dylib naming. In the mean time, do we want to leave it like this or stick it all in ` libexec `?

**Comment by mikemcquaid** [comment]
If we rename the dylibs correctly does Samba still work? Can we try just renaming or deleting the stuff in there?

**Comment by mikemcquaid** [comment]
It could also do with a `test do`.

**Comment by DomT4** [comment]
Will take a look at both things at some point in the next few hours.

**Comment by DomT4** [comment]
Pushed. Had to enormously ` inreplace ` the Makefile to fix the problem, and I fixed the placement of the codepages from ` lib ` into ` share/"codepages" ` whilst I was there. Should eliminate all the audit warnings. Test is kind of crap, but it's the best I can seem to do without it wanting me to connect to something or respond with interactivity in the CLI.

**Comment by vszakats** [comment]
Nice! I'm wondering if just moving `.msg` and `.dat` files to another dir would still keep them picked up by Samba when required?

**Comment by DomT4** [comment]
Should do. It's directly configured in via the ` inreplace `, which propagates successfully to the Makefile:

```
# Where to look for (and install) codepage databases.
CODEPAGEDIR = \/usr/local/Cellar/samba/3.6.25/share/codepages
```

So Samba should know where to find them. Theoretically. It didn't flag up anything wrong when I did some basic testing, but it was only basic testing - I don't use Samba personally so not super-familiar with it.

**Comment by vszakats** [comment]
Looks good, thanks for the input.

**Comment by DomT4** [comment]
Apologies, I misread your initial question. I have my head in a book for uni work and at the same time I glance up every now and again and reply to various threads, but I'm a little ditzy apparently there :smile_cat:.

To actually answer your question: It looks like just ` mv `-ing them elsewhere wouldn't work because the location is hardcoded into both the configure and Makefile scripts.

**Comment by vszakats** [comment]
No worries. It's a pity a fix is non-trivial :( (The reason I was wondering is due to having gone through a similar [issue](https://github.com/vszakats/harbour-core/commit/557cc369b6d09c994134d9ea5b914403a7629701) in another project.

File: Library/Formula/samba.rb [hunk 1]
```diff
@@ -21,11 +21,43 @@ class Samba < Formula
...
+                "LIBTDB_SHARED_TARGET_SONAME=bin/libtdb.1.dylib"
+        s.gsub! "LIBTDB_SHARED_TARGET_FULLVER=bin/libtdb.dylib.1.2.9",
+                "LIBTDB_SHARED_TARGET_FULLVER=bin/libtdb.1.2.9.dylib"
+
+        s.gsub! "libwbclient.dylib.0", "libwbclient.0.dylib"
+        s.gsub! "libaddns.dylib.0", "libaddns.0.dylib"
+        s.gsub! "libnetapi.dylib.0", "libnetapi.0.dylib"
+        s.gsub! "libsmbclient.dylib.0", "libsmbclient.0.dylib"
+        s.gsub! "libsmbsharemodes.dylib.0", "libsmbsharemodes.0.dylib"
+      end
```
Comment (mikemcquaid) [inline Library/Formula/samba.rb]: Just if you fancy a future challenge: I reckon you could do all of these with 1-2 regexes. Happy to pull as-is though.

File: Library/Formula/samba.rb [hunk 2]
```diff
@@ -21,11 +21,43 @@ class Samba < Formula
 
   def install
     cd "source3" do
+      # This stops samba dumping .msg and .dat files directly into lib
+      # It can't be set with a configure switch - There isn't one that fine-grained.
+      inreplace "configure", "${MODULESDIR}", "#{share}/codepages"
```
Comment (mikemcquaid) [inline Library/Formula/samba.rb]: Can you file an issue upstream for this too?

File: Library/Formula/samba.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Formula/samba.rb]:
```ruby
s.gsub! /(lib\w+).dylib(.[\.\d]+)/, "#$1#$2.dylib"
```
:stuck_out_tongue_winking_eye:

Comment (DomT4) [inline Library/Formula/samba.rb]: That ` s.gsub! ` seems to have removed the ` dylib ` name completely :smile_cat:

```ruby
      inreplace "Makefile" do |s|
        s.gsub! /(lib\w+).dylib(.[\.\d]+)/, "#$1#$2.dylib"
      end
```

```
LIBTALLOC_SHARED_TARGET=bin/libtalloc.dylib
LIBTALLOC_SOVER=2
LIBTALLOC_FULLVER=2.0.5
LIBTALLOC_SHARED_TARGET_SONAME=bin/.dylib
LIBTALLOC_SHARED_TARGET_FULLVER=bin/.dylib
```

```
Linking shared library bin/.dylib
ld: library not found for -ltalloc
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [bin/.dylib] Error 1
make: *** Waiting for unfinished jobs....
```

File: Library/Formula/samba.rb (hunk 2 above)
Comment (DomT4) [inline Library/Formula/samba.rb]: Done :+1:

File: Library/Formula/samba.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Formula/samba.rb]:
```ruby
s.gsub! /(lib\w+).dylib(.[\.\d]+)/, "\1\2.dylib"
```

Hope I get right this time.

Comment (DomT4) [inline Library/Formula/samba.rb]: That did something *interesting*.

screen shot 2015-02-26 at 02 06 15

```
Linking shared library bin/.dylib
Linking non-shared library bin/libsmbclient.a
ld: library not found for -ltalloc
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [bin/.dylib] Error 1
make: *** Waiting for unfinished jobs....
```

This is why regexes are the bane of my existence :smile_cat:

Comment (xu-cheng) [inline Library/Formula/samba.rb]: Oh I forget to escape it. 

```ruby
s.gsub! /(lib\w+).dylib(.[\.\d]+)/, "\\1\\2.dylib"
```

Comment (DomT4) [inline Library/Formula/samba.rb]: Yup, that seems to have done the job. Nice one! Cheers :smiley_cat:

**Comment by mikemcquaid** [comment]
Thanks @DomT4!

**Pull request closed without merging**